documentation/source/coroutines.rst

```diff
@@ -202,3 +143,62 @@ type of generator function that can be iterated with ``async for``:
 
 More details on this type of generator can be found in :pep:`525`.
 
+
+.. _yield-syntax:
+
+Generator-based coroutines
+==========================
+
+.. note:: This style is no longer recommended and support may someday be removed.
+
+Prior to Python 3.5, and the introduction of :keyword:`async` and :keyword:`await`, coroutines were implemented as wrappers around generators.
+Coroutine functions would be decorated with :class:`~cocotb.decorators.coroutine` and would use :keyword:`yield` to block on other coroutines or triggers.
+
+.. code-block:: python3
+
+    @cocotb.coroutine
+    def wait_10ns():
+        cocotb.log.info("About to wait for 10 ns")
+        yield Timer(10, units='ns')
+        cocotb.log.info("Simulation time has advanced by 10 ns")
+
+    @cocotb.coroutine
+    def wait_100ns():
+        for _ in range(10):
+            yield wait_10ns()
+
+You may see existing code that uses this syntax for coroutines, but do not worry, it is compatible with async coroutines.
+Any place that will accept async coroutines will also accept generator-based coroutines;
+including :func:`~cocotb.fork`.
+Likewise, any object that can be used in an :keyword:`await` statement can also be used in a :keyword:`yield` statement while in a generator-based coroutine;
+including triggers like :class:`~cocotb.triggers.Timer`.
+Tests can also be generator-based coroutines.
+Tests are not required to be decorated with :class:`~cocotb.decorators.coroutine` as the :class:`~cocotb.decorators.test` decorator will handle this case automatically.
+
+.. code-block:: python3
+
+    async def wait_100ns():
+        for _ in range(10):
+            await wait_10ns()
```

Comment:
```suggestion
            # generator-based coroutines can still be awaited
            await wait_10ns()
```
Also, I think this example belongs after line 272.

Reply:
I'm not sure what line you meant, but I think you mean I could break up the previous paragraph and move the examples after every statement?

Reply:
That's basically what I'm suggesting, yes - make it as clear as possible what each example is demonstrating.

Reply:
Or at least, do it for these two blocks which are already separate.

Reply:
Done.